src/commands/deal/deploy.ts

```diff
@@ -130,6 +131,25 @@ export default class Deploy extends BaseCommand<typeof Deploy> {
       workersConfig,
     });
 
+    const errorMessages = uploadArg.workers.reduce<Array<string>>(
+      (acc, { config: { services }, name }) => {
+        if (services.length === 0) {
+          acc.push(
```

Comment:
I don't like the idea of doing statefull operations in reduce. I would generally prefer to do it like this: `[...acc, newItem]` even though it worse performance-wise. I we really need to push performance to it's limits then I would prefer rewriting in into stateful for loop with explicit mutations.

Also, seems like this code could be rewritten into a chain of `filter` and then `map`. Am I missing something?

Reply:
I did all of these improvements and we will merge it as a part of the next PR. Thanks